### PR TITLE
changed: use AquiferConfig when setting up aquifers

### PR DESCRIPTION
**Pull request by akva2**

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

File: opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp [hunk 1]
```diff
@@ -139,72 +139,32 @@ template <typename TypeTag>
...
-        has = deck.hasKeyword("AQUCT");
-    comm.broadcast(&has, 1, 0);
-
-    if (has) {
-         if (comm.size() > 1)
-             throw std::runtime_error("Aquifers currently do not work in parallel.");
-
-        // updateConnectionIntensiveQuantities();
-        const auto& eclState = this->simulator_.vanguard().eclState();
+    if (aquifer.active()) {
```
Comment (bska) [inline opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp]: I usually prefer
```C++
if (! aquifer.active()) {
    return;
}
```
in these circumstances.  That saves a level of indentation.

File: opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp [hunk 2]
```diff
@@ -139,72 +139,32 @@ template <typename TypeTag>
...
-
-        cartesian_to_compressed_ = cartesianToCompressed(number_of_cells, Opm::UgGridHelpers::globalCell(ugrid));
-
-        for (const auto& aquifer : aquifetp)
-            aquifers_Fetkovich.push_back(AquiferFetkovich<TypeTag>(aquifer_connect[aquifer.aquiferID], cartesian_to_compressed_, this->simulator_, aquifer));
+        const auto& connections = aquifer.connections();
+        for (const auto& aq : aquifer.ct()) {
+            AquiferCarterTracy<TypeTag> act(connections[aq.aquiferID],
+                                            cartesian_to_compressed_, this->simulator_, aq);
+            aquifers_CarterTracy.emplace_back(act);
```
Comment (bska) [inline opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp]: Emplace_back on a fully constructed object is useless.  This should either be
```C++
acquifers_CarterTracey.emplace_back(connections[aq.aquiferID], cartesian_to_compressed_, this->simulator_, aq);
```
or
```C++
aquifer_.push_back(std::move(act));
```
instead

File: opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp [hunk 3]
```diff
@@ -139,72 +139,32 @@ template <typename TypeTag>
...
+        }
 
+        for (const auto& aq : aquifer.fetp()) {
+            AquiferFetkovich<TypeTag> fetp(connections[aq.aquiferID],
+                                           cartesian_to_compressed_, this->simulator_, aq);
+            aquifers_Fetkovich.emplace_back(fetp);
```
Comment (bska) [inline opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp]: Either construct in place or `push_back()` an r-value.

File: opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp (hunk 1 above)
Comment (akva2) [reply opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp]: okay, some like early returns, some do not; changed.

File: opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp (hunk 2 above)
Comment (akva2) [reply opm/simulators/aquifers/BlackoilAquiferModel_impl.hpp]: indeed. this was a complete mess, ended up with something half-way between. to inplace construction.

**Comment by akva2** [comment]
jenkins build this please

**Pull request merged**